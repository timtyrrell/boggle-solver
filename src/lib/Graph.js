import Node from './Node'

class Graph {
  constructor(boardSize, boggleLetters) {
    this.boardSize = Number(boardSize)
    this.boggleLetters = boggleLetters
    this.nodes =  this.createNodes(boardSize, boggleLetters)
  }

  createNodes(boardSize, boggleLetters) {
    const nodes = []

    for (let row = 0; row < boardSize; row++) {
      for (let column = 0; column < boardSize; column++) {
        nodes.push(new Node
          (
            boggleLetters[row][column],
            row,
            column,
            // this.generateConnectedNodeList(this.boggleLetters, row, column, this.boardSize)
          )
        )
      }
    }
    return nodes
  }
}

export default Graph
