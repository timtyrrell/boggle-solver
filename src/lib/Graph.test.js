import React from 'react'
import Graph from './Graph'

const boardSize = '4'
const formattedLetters = [
  ['a','p','p','x'],
  ['x','x','l','x'],
  ['n','x','e','x'],
  ['x','x','x','Z']
]

it('verifies defaults', () => {
  const graph = new Graph(boardSize, formattedLetters)
  expect(graph.boardSize).toEqual(4)
  expect(graph.nodes.length).toEqual(16)
  expect(graph.nodes[0].value).toEqual('a')
  expect(graph.nodes).toMatchSnapshot()
})
