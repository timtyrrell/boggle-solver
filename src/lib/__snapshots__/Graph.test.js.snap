// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verifies defaults 1`] = `
Array [
  Node {
    "value": "a",
  },
  Node {
    "value": "p",
  },
  Node {
    "value": "p",
  },
  Node {
    "value": "x",
  },
  Node {
    "value": "x",
  },
  Node {
    "value": "x",
  },
  Node {
    "value": "l",
  },
  Node {
    "value": "x",
  },
  Node {
    "value": "n",
  },
  Node {
    "value": "x",
  },
  Node {
    "value": "e",
  },
  Node {
    "value": "x",
  },
  Node {
    "value": "x",
  },
  Node {
    "value": "x",
  },
  Node {
    "value": "x",
  },
  Node {
    "value": "Z",
  },
]
`;
